src/js/components/Warning.react.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import React from 'react';
+
+import EditorActions from '../data/EditorActions';
+
+type Props = {
+  activeTab: number,
+  warningSelector: ?string,
+  warnings: any,
+};
+
+type State = {
+  activeWarning: ?number,
+};
+
+class Warning extends React.Component<Props, State> {
```

Comment:
How do you feel about using the pluralized name `Warnings`? That way we can in the future use a `Warning` component for each of the warnings that we display.